### PR TITLE
Upgrade commons-io

### PR DESCRIPTION
**Pull request by mihirgt**

### CODE REVIEW

**Review by kotharironak — DISMISSED** [review]

**Comment by kotharironak** [comment]
Do we use this anywhere?

**Comment by kotharironak** [comment]
> Do we use this anywhere?

Got it.

**Review by kotharironak — APPROVED** [review]

**Review by JBAhire — APPROVED** [review]

File: event-store/build.gradle.kts [hunk 1]
```diff
@@ -16,9 +16,9 @@ tasks.test {
 
 dependencies {
   implementation("com.typesafe:config:1.3.2")
-  implementation(platform("org.hypertrace.core.kafkastreams.framework:kafka-bom:0.2.10"))
+  implementation(platform("org.hypertrace.core.kafkastreams.framework:kafka-bom:0.4.7"))
```
Comment (aaron-steinfeld) [inline event-store/build.gradle.kts:19]: Did we intentionally choose an old version of this? This version of the BOM includes older, vulnerable deps.

File: .github/workflows/build-and-test.yml [hunk 2]
```diff
@@ -8,6 +8,20 @@ on:
   pull_request:
 
 jobs:
+  build:
```
Comment (aaron-steinfeld) [inline .github/workflows/build-and-test.yml:11]: Why did we add this job? It doesn't do anything that the `test` job doesn't already do

**Pull request merged**